docs/api/basic/useSnapshot.mdx

```diff
@@ -7,26 +7,116 @@ description: 'Create a local snapshot that catches changes.'
+
+## Usage
+
+### Read from snapshots in render, use the proxy in callbacks
+
+Snapshots are read-only to render the JSX from their consistent view of the data.
+
+Mutations, and also any reads in callbacks that make mutations, need to be made via the proxy, so that the callback reads & writes the latest value.
 
 ```jsx
 function Counter() {
   const snap = useSnapshot(state)
   return (
     <div>
       {snap.count}
-      <button onClick={() => ++state.count}>+1</button>
+      <button
+        onClick={() => {
+          // also read from the state proxy in callbacks
+          if (state.count < 10) {
+            ++state.count
+          }
+        }}>
+        +1
+      </button>
+    </div>
+  )
+}
+```
+
+### Parent/Child Components
+
+If you have a parent component use `useSnapshot`, it can pass snapshots to child components and the parent & children will re-render when the snapshot changes.
+
+For example:
+
+```jsx
+const state = proxy({
+  books: [
+    { id: 1, title: 'b1' },
+    { id: 2, title: 'b2' },
+  ],
+})
+
+function AuthorView() {
+  const snap = useSnapshot(state)
+  return (
+    <div>
+      {snap.books.map((book) => (
+        <Book key={book.id} book={book} />
+      ))}
+    </div>
+  )
+}
+
+function BookView({ book }) {
+  // book is a snapshot
+  return <div>{book.title}</div>
+}
+```
+
+If book 2's title is changed, a new `snap` is created and the `AuthorView` and `BookView` components will re-render.
+
+Note if `BookView` is `React.memo`d, the 1st `BookView` will not re-render, b/c the 1st `Book` snapshot will be the same instance, as only the 2nd `Book` was mutated (the root `Author` snapshot will also be updated since the list of `books` has changed).
+
+### Child Components Making Mutations
+
+The above approach works if `BookView` is read-only; if your child component needs to make mutations, then you'll need to pass the proxy:
+
+```jsx
+function AuthorView() {
+  const snap = useSnapshot(state)
+  return (
+    <div>
+      {snap.books.map((book, i) => (
+        <Book key={book.id} book={state.books[i]} />
+      ))}
+    </div>
+  )
+}
+
+function BookView({ book }) {
+  // book is the proxy, so we can re-snap it + mutate it
+  const snap = useSnapshot(book)
+  return <div onClick={() => book.updateTitle()}>{snap.title}</div>
+}
+```
+
+Or you can pass both the snapshot and proxy together, if you don't want to call `useSnapshot` in the child component:
+
+```jsx
+function AuthorView() {
+  const snap = useSnapshot(state)
+  return (
+    <div>
+      {snap.books.map((book, i) => (
+        <Book key={book.id} bookProxy={state.books[i]} bookSnapshot={book} />
+      ))}
     </div>
   )
 }
 ```
 
-#### Read only what you need
+There should be no performance difference between these two approaches.
```

Comment:
Do you mean between passing snap and passing proxy state?
Both are render optimized, but the latter with React.memo would perform better with large list. Yeah, but without memo, it's basically no difference (the former might be a bit efficient?). Using memo is kind of advanced, because we can't pass snapshot objects to memoed components. So, maybe it's a different story.